gdnative-core/src/core_types/vector3.rs

```diff
@@ -10,64 +21,441 @@ pub enum Axis {
+    #[inline]
+    pub fn distance_squared_to(self, other: Vector3) -> f32 {
+        other.glam().distance_squared(self.glam())
+    }
+
+    /// Returns the distance to `other`.
+    #[inline]
+    pub fn distance_to(self, other: Vector3) -> f32 {
+        other.glam().distance(self.glam())
+    }
+
+    /// Returns the dot product of this vector and b. This can be used to compare the angle between
+    /// two vectors. For example, this can be used to determine whether an enemy is facing the player.
+    ///
+    /// The dot product will be 0 for a straight angle (90 degrees), greater than 0 for angles
+    /// narrower than 90 degrees and lower than 0 for angles wider than 90 degrees.
+    ///
+    /// When using unit (normalized) vectors, the result will always be between -1.0 (180 degree
+    /// angle) when the vectors are facing opposite directions, and 1.0 (0 degree angle) when the
+    /// vectors are aligned.
+    ///
+    /// Note: a.dot(b) is equivalent to b.dot(a).
+    #[inline]
+    pub fn dot(self, other: Self) -> f32 {
+        self.glam().dot(other.glam())
+    }
+
+    /// Returns a new vector with all components rounded down (towards negative infinity).
+    #[inline]
+    pub fn floor(self) -> Self {
+        Self::gd(self.glam().floor())
+    }
+
+    /// Returns the inverse of the vector. This is the same as
+    /// `Vector3::new(1.0 / self.x, 1.0 / self.y, 1.0 / self.z)`.
+    #[inline]
+    pub fn inverse(self) -> Self {
+        Self::new(1.0 / self.x, 1.0 / self.y, 1.0 / self.z)
+    }
+
+    /// Returns `true` if this vector and v are approximately equal, by running `relative_eq` on
+    /// each component.
+    #[inline]
+    pub fn is_equal_approx(self, v: Self) -> bool {
+        relative_eq!(self.x, v.x) && relative_eq!(self.y, v.y) && relative_eq!(self.z, v.z)
+    }
+
+    /// Returns `true` if the vector is normalized, and `false` otherwise.
+    #[inline]
+    pub fn is_normalized(self) -> bool {
+        self.glam().is_normalized()
+    }
+
+    /// Returns the length (magnitude) of this vector.
+    #[inline]
+    pub fn length(self) -> f32 {
+        self.glam().length()
+    }
+
+    /// Returns the squared length (squared magnitude) of this vector.
+    ///
+    /// This method runs faster than length, so prefer it if you need to compare vectors or need
+    /// the squared distance for some formula.
+    #[inline]
+    pub fn length_squared(self) -> f32 {
+        self.glam().length_squared()
+    }
+
+    /// Returns the result of the linear interpolation between this vector and b by amount t. t is
+    /// on the range of 0.0 to 1.0, representing the amount of interpolation.
+    #[inline]
+    pub fn linear_interpolate(self, b: Self, t: f32) -> Self {
+        Self::gd(self.glam().lerp(b.glam(), t))
+    }
+
+    /// Returns the axis of the vector's largest value. See `Axis` enum.
+    /// If all components are equal, this method returns `Axis::X`.
+    #[inline]
+	#[allow(clippy::collapsible-if)]
```

Comment:
`collapsible_if`. Also this is a tab: run `cargo fmt` before pushing. May I ask why did you ignore instead of use the lint suggestion?

Reply:
I ignore it because I believe the current version is more readable than the collapsed version.

```rust
if self.z > self.y {
    if self.z > self.x {
        Axis::Z
    } else {
        Axis::X
    }
} else {
    if self.y > self.x {
        Axis::Y
    } else {
        Axis::X
    }
}
```

vs

```rust
if self.z > self.y {
    if self.z > self.x {
        Axis::Z
    } else {
        Axis::X
    }
} else if self.y > self.x {
    Axis::Y
} else {
    Axis::X
}
```